### PR TITLE
refactor: Modualize the samples so that they can be gonew

### PR DESCRIPTION
**Pull request by ikawaha**
see. https://github.com/goadesign/examples/issues/131

* Modularized each sample using Go modules.
    *Temporarily replaced the library with `goa.design/examples/basic` the tracing sample. This library replacement is temporary. After this PR is merged and released, we'll need to revert the replacement.

* Update README with gonew cloning instructions. https://github.com/goadesign/examples/pull/130/commits/f700e779f6c40fa7798022234eac3db15183d390

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ This repository contains examples of microservices implemented using
 the directory name. The [cellar](https://github.com/goadesign/examples/tree/master/cellar) example
 provides a complete implementation of a simple microservice.
 
+The samples in each directory serve as templates, and you can clone them using the `gonew` command, as explained in [this blog post](https://go.dev/blog/gonew).
+
+```shell
+$ go install golang.org/x/tools/cmd/gonew@latest
+$ gonew github.com/ikawaha/examples/basic@latest github.com/<your_repo>/basic
```
Comment (raphael) [inline README.md:14]: should this be `gonew goa.design/examples/basic@latest github.com/<your_repo>/basic`?

File: tracing/go.mod [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module goa.design/examples/tracing
+
+go 1.21.3
+
+// Temporarily replacing as the official release is not yet available
+replace goa.design/examples/basic => github.com/ikawaha/calc v0.0.3
```
Comment (raphael) [inline tracing/go.mod:6]: Should this be removed?

File: README.md (hunk 1 above)
Comment (ikawaha) [reply README.md]: ooops, fixed in https://github.com/goadesign/examples/pull/130/commits/e0823af5012140691602c7e782994fe488c4a6e0

File: tracing/go.mod (hunk 2 above)
Comment (ikawaha) [reply tracing/go.mod:6]: Yes, this `replace` directive is temporary until the official release of `goa.design/examples/basic` becomes available. Once the official release is out, we can remove this replacement.

Comment (ikawaha) [reply tracing/go.mod:6]: Would it be okay to remove it in this PR, even if it might cause the tests to fail?

Comment (ikawaha) [reply tracing/go.mod]: To address your feedback regarding the replace directive, I've made adjustments to ensure a smoother build process. Specifically, I've removed the `tracing` sample from this PR (https://github.com/goadesign/examples/pull/130/commits/772a41ac641d5bad0fc12a3b37a49078b0123c8d). Instead, I've prepared a separate PR specifically for adding the tracing sample(https://github.com/ikawaha/examples/pull/1). This eliminates the need to push the noted replace directive. If it works for you, could we merge this PR first and then move on to merging the subsequent PR for `tracing`?

**Comment by ikawaha** [comment]
I've updated the README. I've added information about the Goa Clue example and mentioned the use of the gonew command for easy cloning. [d9ffbdc](https://github.com/goadesign/examples/pull/130/commits/d9ffbdce733e62214d9a59b30f284f5b29747b99)

File: tracing/go.mod (hunk 2 above)
Comment (raphael) [reply tracing/go.mod]: Perfect, this looks great.

**Comment by raphael** [comment]
Thank you for the great contribution!

**Pull request merged**